int32 a
int32 b
bool state_sum
bool state_sub
bool state_mul
bool state_div